java/impl/aws/src/main/java/co/worklytics/psoxy/aws/S3SideOutput.java

```diff
@@ -36,27 +36,29 @@ public S3SideOutput(@Assisted("bucket") String bucket,
             throw new IllegalArgumentException("Bucket name must not be blank");
         }
         this.bucket = bucket;
-        String trimmedPath = StringUtils.trimToEmpty(pathPrefix);
-        this.pathPrefix = (trimmedPath.endsWith("/") || StringUtils.isEmpty(trimmedPath)) ? trimmedPath : trimmedPath + "/";
+        this.pathPrefix = SideOutputUtils.formatObjectPathPrefix(pathPrefix);
     }
 
     @Override
     public void write(@NonNull HttpEventRequest request, @NonNull ProcessedContent content) {
         try {
             AmazonS3 s3Client = s3ClientProvider.get();
 
-            byte[] compressedContent =
-                sideOutputUtils.gzipContent(content.getContent(), content.getContentCharset());
+            //q: correct to do the compression *inside* of every SideOutput implementation?
+            //q: why not have Content-Encoding on ProcessedContent, compress conditionally?
```

Comment:
outdated questions? As finally you've included `contentEncoding` in `ProcessedContent`

Reply:
yeah - I have - thanks.